src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -124,6 +143,18 @@ public boolean isRegularExpression() {
 
     // Returns a regular expression pattern in the form (w1)|(w2)| ... wi are escaped if no regular expression search is enabled
     public Optional<Pattern> getPatternForWords() {
+        return joinWordsToPattern(EscapeMode.JAVA);
+    }
+
+    // Returns a regular expression pattern in the form (w1)|(w2)| ... wi are escaped for javascript if no regular expression search is enabled
+    public Optional<Pattern> getJsPatternForWords() {
```

Comment:
IMHO, although "Js" is quite obvious, I would reword that to getJavaScriptPatternForWords, since you are creating a new Method just for the sole purpose to make its use more obvious in it's name. So why not go all the way?